pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -150,12 +164,41 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		reasonsForWorkerPoolKubernetesUpdate[w.Name] = reasonForWorkerPoolKubernetesUpdate
 	}
 
-	maintainOperation(shoot)
-	maintainTasks(shoot, r.Config)
+	for workerPool, reason := range reasonsForWorkerPoolKubernetesUpdate {
+		operation := fmt.Sprintf("For worker pool %s: %s", workerPool, reason)
+		operations = append(operations, operation)
+	}
+
+	operation := maintainOperation(shoot)
+	if operation != "" {
+		operations = append(operations, operation)
+	}
+
+	tasks := maintainTasks(shoot, r.Config)
+	operations = append(operations, tasks...)
+
+	patch := client.MergeFrom(newShoot.DeepCopy())
+	lastMaintenances = append([]gardencorev1beta1.LastMaintenance{
+		{
+			Operations:    operations,
+			TriggeredTime: metav1.Time{Time: r.Clock.Now()},
+		},
+	}, lastMaintenances...)
```

Comment:
```suggestion
	lastMaintenances = append(lastMaintenances, gardencorev1beta1.LastMaintenance{
		Operations:    operations,
		TriggeredTime: metav1.Time{Time: r.Clock.Now()},
	})
```

Reply:
this is intentional.

Reply:
Oh, for the latest one to be on top. Okay.